Code-4-Community/speak-for-the-trees-frontend
# Connect Leaderboard Routes

## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/925293934/pulses/925416107)

What benefit does this bring to the end user? Or, what benefit does this bring to developers working in the codebase?

Connecting the leaderboard routes populates the leaderboard objects on the volunteer and teams pages with our data, allowing the user to see how who has logged the most trees

## This PR

Describe the changes required and any implementation choices you made to give context to reviewers.

Some of the features looked like they could be abstracted, such as the actions and a lot of the types that the leaderboards use. These are in the actions.ts file and types.ts file in the components/leaderboard/ducks directory. The other ducks files are in the respective container directories for the volunteer and teams leaderboards.

One of the features of the leaderboard pages is the ability to view the best volunteers/teams of the last week/month/year/all-time. Each of these possible time scales requires a different object in the store and each has its own reducer. I think it's possible that a ton of the code here can be abstracted, however I was having a hard time doing so.

## Screenshots

Provide screenshots of any new components, styling changes, or pages.

## Verification Steps

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves.

Unfortunately, it turns out that the axios library we are using cannot accept arguments in request bodies for get requests. This means we are unable to test that the routes pull in data properly.
